filters/src/extract_indices.cpp

```diff
@@ -146,39 +146,52 @@ pcl::ExtractIndices<pcl::PCLPointCloud2>::applyFilter (PCLPointCloud2 &output)
 void
 pcl::ExtractIndices<pcl::PCLPointCloud2>::applyFilter (std::vector<int> &indices)
 {
-  if (negative_)
+  if (indices_->size () > (input_->width * input_->height))
   {
-    // If the subset is the full set
-    if (indices_->size () == (input_->width * input_->height))
-    {
-      // Empty set copy
-      indices.clear ();
-      return;
-    }
+    PCL_ERROR ("[pcl::%s::applyFilter] The indices size exceeds the size of the input.\n", getClassName ().c_str ());
+    indices.clear ();
+    removed_indices_->clear ();
+    return;
+  }
 
-    // Set up the full indices set
-    std::vector<int> indices_fullset (input_->width * input_->height);
-    for (int p_it = 0; p_it < static_cast<int> (indices_fullset.size ()); ++p_it)
-      indices_fullset[p_it] = p_it;
+  if (!negative_)  // Normal functionality
+  {
+    indices = *indices_;
 
-    // If the subset is the empty set
-    if (indices_->empty () || (input_->width * input_->height == 0))
+    if (extract_removed_indices_)
     {
-      // Full set copy
-      indices = indices_fullset;
-      return;
+      // Set up the full indices set
+      std::vector<int> full_indices (input_->width * input_->height);
+      for (int fii = 0; fii < static_cast<int> (full_indices.size ()); ++fii)  // fii = full indices iterator
+        full_indices[fii] = fii;
+
+      // Set up the sorted input indices
+      std::vector<int> sorted_input_indices = *indices_;
+      std::sort (sorted_input_indices.begin (), sorted_input_indices.end ());
+
+      // Store the difference in removed_indices
+      removed_indices_->clear ();
+      set_difference (full_indices.begin (), full_indices.end (), sorted_input_indices.begin (), sorted_input_indices.end (), inserter (*removed_indices_, removed_indices_->begin ()));
```

Comment:
I wonder why compilation does not fail without `std::` for `set_difference` and `inserter`. Did we do `using namespace std` somewhere? :confused: 

Anyway, please add `std::`, we try to always qualify standard library usages. (Same below.)

Reply:
> I wonder why compilation does not fail without std:: for `set_difference` and `inserter`

That'd be ADL (Argument Dependent Lookup) which looks into the namespace domain of the object if there's no other match found. Hence the trick of `using std::Function; Function(stuff)` which allows a better match named `Function` to still replace (read: specialize) the std implementation